libcudacxx/test/atomic_codegen/atomic_add_16b.cu

```diff
@@ -0,0 +1,23 @@
+#include <cuda/atomic>
+
+__global__ void add_relaxed_device_non_volatile(uint16_t* data, uint16_t* out, uint16_t n)
+{
+  auto ref = cuda::atomic_ref<uint16_t, cuda::thread_scope_device>{*(data)};
+  *out     = ref.fetch_add(n, cuda::std::memory_order_relaxed);
+}
+
+/*
+
+; SM8X-LABEL: .target sm_80
+; SM8X:      .visible .entry [[FUNCTION:_.*add_relaxed_device_non_volatile.*]](
+; SM8X-DAG:  ld.param.u64 %rd[[#ATOM:]], [[[FUNCTION]]_param_0];
+; SM8X-DAG:  ld.param.u64 %rd[[#RESULT:]], [[[FUNCTION]]_param_1];
+; SM8X-DAG:  ld.param.u32 %r[[#INPUT:]], [[[FUNCTION]]_param_2];
+; SM8X-NEXT: cvta.to.global.u64 %rd[[#GOUT:]], %rd[[#RESULT]];
+; SM8X-NEXT: //
+; SM8X-NEXT: atom.add.relaxed.gpu.s32 %r[[#DEST:]],[%rd[[#ATOM]]],%r[[#INPUT]];
```

Comment:
This will be incorrect if the value in `data` is, e.g., `numeric_limits<uint16_t>::max()` and `n` is 1, because instead of wrapping around with two's complement, it will be overflowing into the adjacent memory location.

I think this needs to be using `atom.cas.b16` in sm_70+, and `atom.cas.b32` in older GPUs (if supported).

Reply:
It would probably worth it to add "overflow" and "underflow" tests to the integer atomic tests as part of this, for both unsigned and signed integers.

Reply:
Don't worry about this file. It was used for local comparisons. I'm checking for overflow here:
https://github.com/NVIDIA/cccl/pull/2255/files#diff-0c0d40a558c7397280b12a3d7ac94b1e4cd53140e563a96ba70343168c3e5f41